modules/renter/download.go

```diff
@@ -81,8 +81,8 @@ type (
 		siapath           string
 
 		// Syncrhonization tools.
-		downloadFinished chan error
-		mu               sync.Mutex
+		downloadFinished chan struct{}
+		mu               sync.RWMutex
```

Comment:
leaving this as a normal mutex is actually ok. A RWMutex is advantageous only when you're reading a lot, and in parallel. Since we only read in the call to `Err` (which is not called multiple times in parallel), there is little benefit.